src/client/src/admin.rs

```diff
@@ -174,6 +185,12 @@ impl FluvioAdmin {
             .map_err(|err| Error::new(ErrorKind::Other, format!("can't convert: {}", err)).into())
     }
 
+    pub fn watch_topics<'a>(
```

Comment:
I think I know what is wrong.  Instead of implementing watch Api here, we should just delegate it. so it should look like this;
```
pub fn watch<s>(&mut self) -> impl Stream<Item = MetadataChanges<TopicSpec, AlwaysNewContext>>  {
     self.get_metadata<S>().watch()
}
```

Reply:
Other option is to put this API under `unstable` features since this may be changed in the future.

Reply:
How would you go about implementing `get_metadata<S>()`? The `MetadataStores` holds only three `StoreContext`s, topic, partition, and spu. So this function cannot be defined for all `S: Spec`, rather it can only work for TopicSpec, PartitionSpec, and SpuSpec. Would we need to create a new trait for marking the Specs we can use? That seems like a clunky approach. Another option is to switch on TypeId's but that is also gross. The current quick-and-dirty thing I chose to do was to specialize the methods and have watch_topic, watch_partition, and watch_spu, but that is also really not great. I'd like to hear other thoughts on the API design, for now I'm going to try to get the stream actually working.